src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -1543,6 +1558,10 @@ - (void)_handleReportBegin
         [self _changeState:MTRDeviceStateReachable];
     }
 
+    // Reset _timeSynchronizationLossDetected, so that it will get set based
+    // on the values in this report.
+    _timeSynchronizationLossDetected = NO;
+
```

Comment:
Holding a comment here, so @bzbarsky-apple can resolve when he wants this to merge